### PR TITLE
Allow Periph to compile with !AP_GPS_ENABLED

### PR DESCRIPTION
**Pull request by peterbarker**

### CODE REVIEW

File: libraries/AP_Common/Location.cpp [hunk 1]
```diff
@@ -48,6 +49,7 @@ Location::Location(const Vector3f &ekf_offset_neu, AltFrame frame)
         offset(ekf_offset_neu.x * 0.01, ekf_offset_neu.y * 0.01);
     }
 }
+#endif
```
Comment (tridge) [inline libraries/AP_Common/Location.cpp:52]: // AP_AHRS_ENABLED

File: libraries/AP_Common/Location.cpp [hunk 2]
```diff
@@ -156,6 +163,9 @@ bool Location::get_alt_cm(AltFrame desired_frame, int32_t &ret_alt_cm) const
                 alt_abs = alt + ekf_origin.alt;
             }
             break;
+#else
+            return false;
+#endif
```
Comment (tridge) [inline libraries/AP_Common/Location.cpp:168]: //endif

File: libraries/AP_Common/Location.cpp (hunk 1 above)
Comment (peterbarker) [reply libraries/AP_Common/Location.cpp]: Done

File: libraries/AP_Common/Location.cpp (hunk 2 above)
Comment (peterbarker) [reply libraries/AP_Common/Location.cpp]: Done

**Comment by peterbarker** [comment]
... added a patch that creates the AHRS singleton if AP_AHRS_ENABLED is true, which would stop HerePro from dying if you were to call AHRS methods on it.

CubeOrange-Periph-heavy got 880 bytes smaller as it doesn't include scripting and thus no AHRS is required.

**Review by tridge — APPROVED** [review]

**Pull request merged**